content_script.js

```diff
@@ -1,82 +1,213 @@
+    } else if (element.tagName && this.FILTERED_TAGS.has(element.tagName.toLowerCase())) {
+      // filter out elements that are in filteredTags
+    } else if (element.nodeType == Node.TEXT_NODE) {
+      this.html.push(element.textContent);
+    } else {
+      this.html.push(new Array(depth+1).join('  '));
+      this.html.push(`<${element.tagName.toLowerCase()} `);
+
+      var style = element.ownerDocument.defaultView.getComputedStyle(element, null).cssText;
+      this.styleIndices.push(this.html.length);
+      this.html.push(`style="${style}" `);
+
+      var attributes = element.attributes;
+      if (attributes) {
+        for (var i = 0, attribute; attribute = attributes[i]; i++) {
+          if (attribute.name.toLowerCase() == 'src') {
+            if (element.tagName.toLowerCase() != 'iframe') {
+              this.html.push(`${attribute.name}=`);
+              this.html.push(''); // entry where properly escaped quotes will go
+              this.srcHoles[this.html.length] = attribute.value; // mark location to insert url later
+              this.html.push(''); // entry where data url will go
+              this.html.push(''); // entry where properly escaped quotes will go
+              this.html.push(' ');
+            }
+          } else if (attribute.name.toLowerCase() != 'style') {
+            this.html.push(`${attribute.name}="${attribute.value}" `);
+          }
         }
-        else if (attribute.name.toLowerCase() != "style") {
-          html.push(attribute.name + '="' + attribute.value + '" ');
+        if (element.tagName.toLowerCase() == 'iframe') { // TODO: IS IT POSSIBLE AN IFRAME CAN HAVE NO ATTRIBUTES?
+          this.html.push('srcdoc=');
+          this.frameHoles[this.html.length] = this.iframeFullPosition(window) + '.' + this.iframePosition(element.contentWindow);
+          this.html.push(''); // entry where the iframe contents will go
+          this.html.push(' ');
         }
       }
-      html.push(">\n");
+
+      this.html.push('>\n');
+
+      var children = element.childNodes;
+      if (children) {
+        for (var i = 0, child; child = children[i]; i++) {
+          this.serializeTree(child, depth+1);
+        }
+      }
+
+      this.html.push(new Array(depth+1).join('  '));
+      this.html.push(`</${element.tagName.toLowerCase()}>\n`);
     }
-    
-    var children = element.childNodes;
-    if (children) {
-      for (var i = 0; i < children.length; i++) {
-        var child = children[i];
-        html.push(serializeTree(html, holes, child, depth+1));
+  }
+
+  /**
+   * Takes an html document, and populates this objects fields such that it can eventually be converted into an
+   * html file.
+   *
+   * @param {Document} doc The Document to serialize. Defaults to the current document
+   * @public
+   */ 
+  serializeDocument(doc) {
+    doc = doc || document;
+    this.html.push('<!DOCTYPE html>\n');
+    var node = doc.firstChild;
```

Comment:
The goal is to find the first child node of the document that isn't a Node.DOCUMENT_TYPE_NODE (i.e. isn't <!DOCTYPE html> or something similar).  I've encountered frames that don't start with a Node.DOCUMENT_TYPE_NODE, so this loop was simply to find the first non Node.DOCUMENT_TYPE_NODE.  Perhaps it's overkill because there will never be more than 1?